libbeat/docs/outputconfig.asciidoc

```diff
@@ -527,9 +565,57 @@ The password for connecting to Kafka.
 
 ===== topic
 
-The Kafka topic used for produced events. The setting can be a format string
+The Kafka topic used for produced events. The setting can be a *Format String*
 using any event field. To set the topic from document type use `%{[type]}`.
 
+===== topics
+
+Array of topic selector rules supporting conditionals, *Format String*
+based field access and name mappings. The first rule matching will be used to
+set the `topic` for the event to be published. If `topics` is missing or no
+rule matches, the `topic` field will be used.
+
+Rule settings:
+
+*`topic`*: The topic *Format String* to use. If the fields used are missing, the
+ rule fails.
+
+*`mapping`*: Dictionary mapping index names to new names
+
+*`default`*: Default string value if `mapping` does not find a match.
+
+*`when`*: Condition which must succeed in order to execute the current rule.
+
+===== key
+
+The Kafka event key. The event key must be unique and can be extracted from the
+event using a *Format String*.
+
+===== partition
+
+Kafka output broker event partitioning strategy. Must be one of `random`,
+`round_robin`, or `hash`. By default the `hash` partitioner is used.
+
+*`random.group_events`*: Select a new partition by random every `group_events`
```

Comment:
That sort of substitution works better when the name of the option clearly indicates what it represents. In this case, it's not clear by the name that `group_events` will resolve to a number, so some readers will stumble over how the sentence is constructed. I realize you're trying to avoid using a lot of words, but it's better to be explicit like you've been in your explanation to me. Also, unless you're telling the user to do something, avoid using the imperative. When someone reads "select", they think they need to select something. It might take them a few seconds to realize that you're essentially saying "configure this option so that the software selects."  Does that make sense?